src/main/java/core/basesyntax/writer/impl/WriterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.writer.impl;
+
+import core.basesyntax.writer.Writer;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+public class WriterImpl implements Writer {
+    private static final String SPLIT_GAP = " ";
+
+    @Override
+    public void write(String report, String path) {
+        String[] split = report.split(SPLIT_GAP);
+        try (BufferedWriter writer = new BufferedWriter(new java.io.FileWriter(path))) {
+            for (String s : split) {
+                writer.write(s);
```

Comment:
The report is being split by spaces, which may not preserve the intended format of the report. Consider writing the entire report string to the file without splitting it. This will ensure that the report's formatting, such as line breaks, is preserved.